doc/advanced-configuration/outgoing-connections.md

```diff
@@ -1,177 +1,92 @@
+* `generic` - pool of generic workers not associated directly with a particular connection
 
-Every entry in the `outgoing_pools` is a 5-element tuple:
+* **Syntax:** Each pool is specified in a subsection starting with `[outgoing_pools.type.tag]`, where `type` is one of available connection types and `tag` is an arbitrary value.
+This allows you to create multiple dedicated pools of the same type.
 
-```erlang
-{Type, Host, Tag, PoolOptions, ConnectionOptions}
-```
+# General pool options
 
-Where:
+#### `outgoing_pools.*.*.scope`
+* **Syntax:** string. Allowed values: `"global"`, `"host"`, `"single_host"`
+* **Default:** `"global"`
+* **Example:** `scope = "host"`
 
-* `Type` is one of the types listed above
-* `Host` can be set to:
-    * `global` - meaning the pool will started once no matter how many XMPP hosts served by MongooseIM
-    * `host` - the pool will be started for all the XMPP hosts served by MongooseIM
-    * a binary representing a specific XMPP host like `<<"domain1.chat.im">>`
-* `Tag` is a name to distinguish pools with the same `Type` and `Host` parameter.
-* `PoolOptions` is a list of `{key, value}` pairs as defined in [worker_pool doc](https://github.com/inaka/worker_pool#starting-a-pool)
-   with the following exception:
-    * `strategy` - specifies the worker selection strategy for the given pool, default is `best_worker`,
-      more details on this can be found in [Choosing strategy in worker_pool doc](https://github.com/inaka/worker_pool#choosing-a-strategy)
-      *WARNING:* `redis` and `riak` backends are not compatible with `available_worker` strategy.
-    * `call_timeout` - specifies the timeout, in milliseconds, for a call operation to the pool
-* `ConnectionOptions` - options list passed to the `start` function of the pool type
+#### `outgoing_pools.*.*.host`
+* **Syntax:** string
+* **Default:** no default; required if `"single_host"` scope is specified
+* **Example:** `host = "anotherhost.com"`
 
+`scope` can be set to:
+* `global` - meaning that the pool will started once no matter how many XMPP hosts are served by MongooseIM
+* `host` - the pool will be started for all the XMPP hosts served by MongooseIM
+* `single_host` - the pool will be started for the selected host only (you must provide host name).
 
-### Examples
+# Worker pool options
 
-Provided MongooseIM serves domains `<<"a.com">>`, `<<"b.com">>`, `<<"c.eu">>` and `<<"d.eu">>`
-the following `outgoing_pools` configuration:
+All pools are managed by [inaka/worker_pool](https://github.com/inaka/worker_pool) library.
 
-```erlang
-{redis, <<"a.com">>, default, PoolOpts, ConnOptsForDomain1},
-{redis, host, default, PoolOpts, ConnOpts},
-{redis, <<"c.eu", default, PoolOpts, ConnOptsForDomain2}
-```
+Available options are:
+#### `outgoing_pools.*.*.strategy`
+* **Syntax:** `"best_worker"`, `"random_worker"`, `"next_worker"`, `"available_worker"` or `"next_available_worker"`
+* **Default:** `"available_worker"`
```

Comment:
I'm sure it depends on the pool type... and I think we should notify the user about it.

Reply:
From what I see it doesn't, but it is indeed different - available_worker is default in wpool, but mongoose_wpool makes it best_worker